### PR TITLE
Feature/uche/add items profile

### PR DESCRIPTION
**Pull request by enyeInc**
- the add items view is now selected from the user profile side menu

### CODE REVIEW

File: src/components/app/components/index.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import App from './App';
+import AppLink from './AppLink';
+
 import './styles.css';
 
-export { App };
+export { App, AppLink };
```
Comment (Simpcyclassy) [inline src/components/app/components/index.js]: Why was this module not exported before?

File: src/components/userProfile/components/SideMenu.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { Menu, Icon } from 'antd';
 
+import { AppLink } from '../../app/components';
```
Comment (Simpcyclassy) [inline src/components/userProfile/components/SideMenu.js]: Oh! I see...

File: src/components/router/components/Routes.js [hunk 3]
```diff
@@ -18,14 +17,8 @@ const Routes = () => (
...
                 <Route path={MARKET} exact strict component={home.components.Home} />
-                <Route path={PROFILE} exact strict component={userProfile.components.UserProfile} />
-                <Route
-                    path={ADD_ITEM}
-                    exact
-                    strict
-                    component={addItem.components.AddItem}
-                />
-                <Route path={WHO_WE_ARE} exact strict component={aboutUs.components.AboutUs} />
+                <Route path={PROFILE} component={userProfile.components.UserProfile} />
```
Comment (Simpcyclassy) [inline src/components/router/components/Routes.js]: Why did you remove `exact` and `strict`

File: src/components/userProfile/components/SideMenu.js [hunk 4]
```diff
@@ -15,11 +16,18 @@ export default class SideMenu extends React.Component {
                 className={MENU}
                 theme={DARK}
                 mode={INLINE}
+                selectedKeys={[location.pathname]}
```
Comment (Simpcyclassy) [inline src/components/userProfile/components/SideMenu.js]: is `pathname` a built in method?

File: src/components/userProfile/components/SideMenu.js [hunk 5]
```diff
@@ -15,11 +16,18 @@ export default class SideMenu extends React.Component {
...
-                            <span><Icon type={iconType} />{label}</span>
+                    SIDE_MENU_ITEMS.map(({ iconType, label, link }) => (
+                        <Menu.Item key={link || label}>
+                            {
+                                link ?
+                                    <AppLink to={link}>
+                                        <span><Icon type={iconType} />{label}</span>
+                                    </AppLink> :
+                                    <span><Icon type={iconType} />{label}</span>
+                            }
```
Comment (Simpcyclassy) [inline src/components/userProfile/components/SideMenu.js]: When all the side menu items have a link would it still be necessary to use ternary?

File: src/components/userProfile/components/UserProfile.js [hunk 6]
```diff
@@ -12,19 +15,29 @@ const {
...
                     <Sider width={200} className={SIDER}>
                         <SideMenu />
                     </Sider>
                     <Content className={SM_CONTENT}>
-                        <ProfileContent />
+                        <Route path={'/profile'} exact strict component={ProfileContent} />
+                        <Route
+                            path={`${match.path}/items`}
+                            component={AddItem}
+                        />
```
Comment (Simpcyclassy) [inline src/components/userProfile/components/UserProfile.js]: I see what you did here.

File: src/components/userProfile/components/SideMenu.js (hunk 5 above)
Comment (enyeInc) [reply src/components/userProfile/components/SideMenu.js]: nope it would not be

File: src/components/userProfile/components/SideMenu.js (hunk 4 above)
Comment (enyeInc) [reply src/components/userProfile/components/SideMenu.js]: The real question here is, what is [`location`](https://developer.mozilla.org/en-US/docs/Web/API/Location)? [`location`](https://developer.mozilla.org/en-US/docs/Web/API/Location) is a global object and `pathname` is a key on it that points to the exact path that your page is on. Your browser uses this [`location`](https://developer.mozilla.org/en-US/docs/Web/API/Location) object to keep track of the different parts of a `URL`.

File: src/components/router/components/Routes.js (hunk 3 above)
Comment (enyeInc) [reply src/components/router/components/Routes.js]: screen shot 2019-02-03 at 9 54 20 am
Per the documentation, if i want to have other paths that match `/profile` and display within the `UserProfile` component (ex: `/profile/items`) i have to remove exact. Given how our routing is set up, all requests to anything starting with `/profile` will only match to `Userprofile` and nothing else. Secondly, `strict` isn't really needed here, it has no affect because we have additional url segments in the pathname. I actually think we can get rid of it for most of the routes, but not exactly sure.

File: src/components/userProfile/components/UserProfile.js (hunk 6 above)
Comment (enyeInc) [reply src/components/userProfile/components/UserProfile.js]: `match` is added to any component that is displayed using `react router`. Its very similar to the global object `location` but the only difference is that it is unique to the component. It tells use the current path and other information associated with the components location in the browser. I am just leveraging it in order to make the app and codebase more dynamic. I could have easily hard coded the `items` pathname as such:
```
                        <Route
                            path={`/profile/items`}
                            component={AddItem}
                        />
```
However, this way if we decided to changed the path to `UserProfile` to lets say `/account` then we would not need to change anything for this piece of code to still work. The new path `/account/items` will automatically be picked up.

File: src/components/userProfile/components/SideMenu.js (hunk 4 above)
Comment (Simpcyclassy) [reply src/components/userProfile/components/SideMenu.js]: Oh I see...

**Comment by Simpcyclassy** [comment]
Looks good to me except there seems to be a conflict that needs to be resolved.

**Comment by EngrEric** [comment]
resolve the conflicts at package.json and its good to me

**Comment by joshuaobinna** [comment]
Looks good to me
